### ISSUE TITLE
[rl] QR-DQN huber(): eagerly-evaluated masked-out branch turns 0 · (−inf) into NaN

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#888]
## The bug

`huber()` blends its two branches arithmetically, so the **untaken** branch is still evaluated and can poison the result.

`crates/rlevo-reinforcement-learning/src/algorithms/qrdqn/quantile_loss.rs:26-33`:
```rust
pub fn huber<B: Backend, const D: usize>(u: Tensor<B, D>, kappa: f32) -> Tensor<B, D> {
    let abs_u = u.clone().abs();
    let small = abs_u.clone().lower_equal_elem(kappa).float();
    let quad = u.powi_scalar(2).mul_scalar(0.5);
    let linear = abs_u.sub_scalar(0.5 * kappa).mul_scalar(kappa);
    small.clone() * quad + small.neg().add_scalar(1.0) * linear
}
```

`linear` is computed for **every** element regardless of the mask. When `0.5·κ²` overflows f32, `linear` becomes `−inf`, and the discard step is `0.0 · (−inf)` = **NaN** — in every element, including those the mask says should be purely quadratic. IEEE-754 has no zero-annihilates-infinity rule, so masking by multiplication cannot discard a poisoned operand.

## Measured

`Flex` backend, `quantile_huber_loss_per_sample`, 1×3 pred/target, Apple Silicon / macOS 25.5.0:

| κ | loss | finite? |
|---|---|---|
| 1.0 | 2.7083334e-1 | yes |
| 1e18 | 2.7833335e-19 | yes |
| 2.6e19 | 1.0705128e-20 | yes |
| **2.6087635e19** | 1.0669165e-20 | yes |
| **2.6087637e19** (next f32) | **NaN** | no |
| 1e20 | NaN | no |
| `f32::MAX` | NaN | no |

The cutoff is exactly `√(2·f32::MAX) = 2.6087636e19`, confirmed by bit-level binary search — the last accepted `f32` produces a finite loss and the very next one is NaN.

## Why #345 does not close this

#345 added a config-boundary guard rejecting κ where `0.5·κ²` is non-finite, which closes the path *for configs*. It does not close the function:

`quantile_huber_loss_per_sample` is `pub fn` (`quantile_loss.rs:79`) in `pub mod quantile_loss` → `pub mod qrdqn` → `pub mod algorithms` (`lib.rs:71-78`). Any downstream crate can call it directly:

```rust
use rlevo_reinforcement_learning::algorithms::qrdqn::quantile_loss::quantile_huber_loss_per_sample;
let loss = quantile_huber_loss_per_sample(&pred, &target, &taus, 1e20); // -> NaN
```

`kappa` has no `assert`, no `debug_assert`, and no documented precondition on this path. `huber` itself is public too and has the same exposure.

Note the guard in #345 is phrased as *"the linear branch must not overflow"* precisely because it is compensating for this implementation detail from the outside. If `huber` stops eagerly evaluating the untaken branch, that config bound becomes unnecessary and should be relaxed — the two are coupled, which is itself an argument for fixing it here.

## Suggested fix

Select instead of blend. `mask_where` discards the unselected operand rather than multiplying it by zero, so a poisoned untaken branch cannot contaminate the result. There is in-repo precedent: `ppo/losses.rs:173-181` uses `mask_where` and is safe against exactly this.

Verify numerical equivalence against the current implementation for valid κ before/after — this is loss-function code and the change must be bit-identical on the healthy path.

Also worth doing regardless: document κ's precondition on both public functions, and consider a `debug_assert!(kappa.is_finite() && kappa > 0.0)`.

## Severity

Low in practice today, and deliberately not filed higher. Nothing in this repo passes a bad κ — the only direct caller is `benches/qrdqn_bench.rs:62` with a hardcoded `1.0`, and `rlevo-benchmarks*` has zero references. `FiniteLossGuard` (ADR 0056, #318) also catches a non-finite loss at `qrdqn_agent.rs:573` before `backward()` for anyone going through the agent, so weights are never corrupted — the symptom is a stalled run, not silent poisoning.

The exposure is to external callers of the public API, and the value is in removing a footgun from a `pub` numerics function rather than fencing it off from one direction.

## Provenance

Found during the QA pass on #345. The κ = `+∞` half of #345 was a symptom of this; the huge-but-finite half was found by sweeping the domain rather than testing the boundary the issue named. Related: #440 (item 6, `Kappa` newtype — same public-API gap, different remedy).
